containers/ecr-viewer/src/app/services/listConditionsService.ts

```diff
@@ -12,38 +14,59 @@ export const getAllConditions = async (): Promise<string[]> => {
   if (!user) return [];
 
   try {
-    const result = await getDb<Core>()
-      .selectFrom("ecr_rr_conditions")
-      .select("condition")
-      .distinct()
-      .orderBy("condition")
-      .$if(user.user_type !== "admin", (qb) =>
-        qb
-          .innerJoin(
-            "condition_reference",
-            "condition_reference.code",
-            "ecr_rr_conditions.condition_code",
-          )
-          .innerJoin(
-            "program_area",
-            "program_area.uuid",
-            "condition_reference.program_area_uuid",
-          )
-          .innerJoin(
-            "user_program_area",
-            "user_program_area.program_area_uuid",
-            "program_area.uuid",
-          )
-          .where("user_program_area.user_uuid", "=", user.uuid),
-      )
-      .execute();
-    return result.map((row) => row.condition);
+    const db = getDb<Core>();
+    const result = await db.transaction().execute(async (transaction) => {
+      const conditionsResult = await transaction
+        .selectFrom("ecr_rr_conditions")
+        .select("condition")
+        .distinct()
+        .orderBy("condition")
+        .$if(user.user_type !== "admin", (qb) =>
+          qb
+            .innerJoin(
+              "condition_reference",
+              "condition_reference.code",
+              "ecr_rr_conditions.condition_code",
+            )
+            .innerJoin(
+              "program_area",
+              "program_area.uuid",
+              "condition_reference.program_area_uuid",
+            )
+            .innerJoin(
+              "user_program_area",
+              "user_program_area.program_area_uuid",
+              "program_area.uuid",
+            )
+            .where("user_program_area.user_uuid", "=", user.uuid),
+        )
+        .execute();
+
+      const hasNoConditionsResult = await transaction
+        .selectFrom("ecr_data")
+        .leftJoin(
+          "ecr_rr_conditions",
+          "ecr_data.eicr_id",
+          "ecr_rr_conditions.eicr_id",
+        )
+        .select("ecr_data.eicr_id")
+        .where("ecr_rr_conditions.uuid", "is", null)
+        .executeTakeFirst();
+
+      const actualConditions = conditionsResult.map((row) => row.condition);
+      const hasEcrWithNoConditions = hasNoConditionsResult !== undefined;
+
+      if (hasEcrWithNoConditions) {
+        return [NO_CONDITIONS_REPORTED_OPTION, ...actualConditions];
+      }
+      return actualConditions;
+    });
+    return result;
```

Comment:
return from the start instead of the end? I think it'll be more readable that way